sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -401,10 +422,75 @@ def analyze_property(
                 ref=ref,
                 optional=optional,
                 plain=plain,
+                description=self.get_docstring(typ.__name__, name),
             )
         else:
             raise ValueError(f"unsupported type {arg}")
 
+    def find_docstrings(self, path: Path) -> dict[str, dict[str, str]]:
+        """
+        find_docstrings returns the docstrings for all the attributes of all
+        the classes in `self.path.
+
+        Unfortunately, only class docstrings are available at runtime, the
+        docstrings of the attributes are not available. Instead of relying on
+        runtime information we parse the source code to extract the docstrings.
+        """
+        docs: dict[str, dict[str, str]] = {}
+        for file_path in self.iter(path):
+            if file_path.suffix != ".py":
+                continue
+            with open(file_path) as f:
+                src = f.read()
+                t = ast.parse(src)
+                docs.update(self.find_docstrings_in_module(t))
+        return docs
+
+    def find_docstrings_in_module(self, mod: ast.Module) -> dict[str, dict[str, str]]:
```

Comment:
Neat, I would have thought this would be harder!

Reply:
Same :D